index incremented by 1 when value is greater than 1.

**Describe the bug**
even though the key is using `dependencies[2].version` the application crashes with:

```
panic: runtime error: index out of range [3] with length 3

goroutine 1 [running]:
github.com/olblak/updateCli/pkg/plugins/yaml.replace(0xc000b7c140, 0xc0006cad30, 0x1, 0x1, 0xc00818b5a0, 0x5, 0x1, 0x4f9dd7, 0xc0004afe40, 0x17, ...)
        /home/runner/work/updatecli/updatecli/pkg/plugins/yaml/main.go:198 +0xc0b
github.com/olblak/updateCli/pkg/plugins/yaml.replace(0xc000b7c000, 0xc0006cad20, 0x2, 0x2, 0xc00818b5a0, 0x5, 0x1, 0x2, 0x2, 0x43f16f, ...)
        /home/runner/work/updatecli/updatecli/pkg/plugins/yaml/main.go:249 +0x7cd
github.com/olblak/updateCli/pkg/plugins/yaml.(*Yaml).TargetFromSCM(0xc000880a80, 0xc00818b5a0, 0x5, 0x1961620, 0xc000758960, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /home/runner/work/updatecli/updatecli/pkg/plugins/yaml/target.go:122 +0x2ce
github.com/olblak/updateCli/pkg/core/engine/target.(*Target).Run(0xc00057ce68, 0xc00818b5a0, 0x5, 0x248e3d8, 0xc000216bc8, 0xc000b72360, 0x2c)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/target/main.go:108 +0x32c
github.com/olblak/updateCli/pkg/core/engine.RunTargets(0xc00057da40, 0x248e3d8, 0xc00042e2d0, 0xc0004af6c0, 0xe, 0x9)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/main.go:389 +0x4d3
github.com/olblak/updateCli/pkg/core/engine.(*Engine).Run(0x248e3a0, 0x0, 0x0)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/main.go:273 +0xeac
github.com/olblak/updateCli/cmd.run(0x17000ab, 0x4)
        /home/runner/work/updatecli/updatecli/cmd/root.go:71 +0x174
github.com/olblak/updateCli/cmd.glob..func2(0x2479240, 0xc0001a9e40, 0x0, 0x4)
        /home/runner/work/updatecli/updatecli/cmd/diff.go:27 +0x149
github.com/spf13/cobra.(*Command).execute(0x2479240, 0xc0001a9e00, 0x4, 0x4, 0x2479240, 0xc0001a9e00)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x2479780, 0x43, 0xc00036bf48, 0x44430a)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/olblak/updateCli/cmd.Execute()
        /home/runner/work/updatecli/updatecli/cmd/root.go:35 +0x31
main.main()
        /home/runner/work/updatecli/updatecli/main.go:8 +0x20
make: *** [Makefile:7: updatecli.diff] Error 2
```

`apps/influxdb/Chart.yaml`
```yaml
apiVersion: v2
appVersion: "1.0"
description: Proxy-chart for influxdata
name: influxdb
version: 1.0.0
dependencies:
  - name: influxdb
    version: 4.8.9
    repository: https://helm.influxdata.com/
  - name: chronograf
    version: 1.1.18
    repository: https://helm.influxdata.com/
  - name: kapacitor
    version: 1.3.1
    repository: https://helm.influxdata.com/

```

`kapicator.yaml.tpl`
```yaml
source:
  kind: helmChart
  spec:
    url: https://helm.influxdata.com/
    name: kapacitor

targets:
  chartVersion:
    name: "kapacitor Helm Chart"
    kind: yaml
    spec:
      file: "apps/influxdb/Chart.yaml"
      key: "dependencies[2].version"
    scm:
      github:
        user: "{{ .github.user }}"
        email: "{{ .github.email }}"
        owner: "{{ .github.owner }}"
        repository: "{{ .github.repository }}"
        token: "{{ requiredEnv .github.token }}"
        username: "{{ .github.username }}"
        branch: "{{ .github.branch }}"
```

**Version (please complete the following information):**
 - OS: linux
 - Version v0.0.27

--- a/pkg/plugins/yaml/main.go
+++ b/pkg/plugins/yaml/main.go
@@ -185,11 +185,16 @@ func replace(entry *yaml.Node, keys []string, version string, columnRef int) (fo
 				column = entry.Content[index+1].Column
 
 				if entry.Content[index+1].Kind == yaml.SequenceNode && entry.Content[index+1].Content[0].Kind == yaml.MappingNode {
-					positionIndex := 0
+					/*
+						If looking for a key at a specific position in a sequence like keys[2].key
+						for following example should parse - key : c
+						keys:
+							- key: a
+							- key: b
+							- key: c
+					*/
 
-					if (position*2)-1 >= 0 {
-						positionIndex = (position * 2) - 1
-					}
+					positionIndex := position
 
 					if positionIndex > len(entry.Content[index+1].Content) {
 						return false, "", 0

--- a/pkg/plugins/yaml/main_test.go
+++ b/pkg/plugins/yaml/main_test.go
@@ -22,10 +22,16 @@ image3:
   version: 1.0
 - name: b
   version: 1.5
+- name: c
+  version: 3.5
+- name: d
+  version: 5.5
 image4:
 - c
 - d
 - f
+- g
+- h
 image5::tag: 1.17
 image6::tags: 
 - 1.17
@@ -230,6 +236,16 @@ func TestReplace(t *testing.T) {
 			expectedValueFound: true,
 		},
 		{
+			key:                []string{"image3[2]", "version"},
+			expectedOldVersion: "3.5",
+			expectedValueFound: true,
+		},
+		{
+			key:                []string{"image3[3]", "version"},
+			expectedOldVersion: "5.5",
+			expectedValueFound: true,
+		},
+		{
 			key:                []string{"image5::tag"},
 			expectedOldVersion: "1.17",
 			expectedValueFound: true,
@@ -247,6 +263,16 @@ func TestReplace(t *testing.T) {
 		{
 			key:                []string{"image4[1]"},
 			expectedOldVersion: "d",
+			expectedValueFound: true,
+		},
+		{
+			key:                []string{"image4[2]"},
+			expectedOldVersion: "f",
+			expectedValueFound: true,
+		},
+		{
+			key:                []string{"image4[3]"},
+			expectedOldVersion: "g",
 			expectedValueFound: true,
 		},
 		{

